File: src/AngelissimaApi/Models/Product.cs
namespace AngelissimaApi.Models
{
    using System.Collections.Generic;

    public class Product : BaseItem
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal SalePrice { get; set; }
        public int RecommendedQuantity { get; set; }
        public IEnumerable<BarCode> BarCodes { get; set; }
    }
}

File: app/src/app/products/shared/product.ts
import { BarCode } from '../../barcodes/shared/barcode';

export class Product {
  productId: number;
  name: string;
  description: string;
  unitPrice: number;
  salePrice: number;
  recommendedQuantity: number = 1;
  barCodes: BarCode[];
}